pkg/ingester/metrics.go

```diff
@@ -676,3 +678,69 @@ func (sm *tsdbMetrics) setRegistryForUser(userID string, registry *prometheus.Re
 func (sm *tsdbMetrics) removeRegistryForUser(userID string) {
 	sm.regs.RemoveUserRegistry(userID, false)
 }
+
+type matcherCacheMetrics struct {
+	r *prometheus.Registry
+
+	requestsTotal *prometheus.Desc
+	hitsTotal     *prometheus.Desc
+	numItems      *prometheus.Desc
+	maxItems      *prometheus.Desc
+	evicted       *prometheus.Desc
+}
+
+func newMatchCacheMetrics(r *prometheus.Registry) *matcherCacheMetrics {
+	m := &matcherCacheMetrics{
+		r: r,
+		requestsTotal: prometheus.NewDesc(
+			"ingester_matchers_cache_requests_total",
+			"Total number of cache requests for series matchers",
+			nil, nil),
+		hitsTotal: prometheus.NewDesc(
+			"ingester_matchers_cache_hits_total",
+			"Total number of cache hits for series matchers",
+			nil, nil),
+		numItems: prometheus.NewDesc(
+			"ingester_matchers_cache_items",
+			"Total number of cached items",
+			nil, nil),
+		maxItems: prometheus.NewDesc(
+			"ingester_matchers_cache_max_items",
+			"Maximum number of items that can be cached",
+			nil, nil),
+		evicted: prometheus.NewDesc(
+			"ingester_matchers_cache_evicted_total",
+			"Total number of items evicted from the cache",
+			nil, nil),
+	}
+	return m
+}
+
+func (m *matcherCacheMetrics) Describe(out chan<- *prometheus.Desc) {
+	out <- m.requestsTotal
+	out <- m.hitsTotal
+	out <- m.numItems
+	out <- m.maxItems
+	out <- m.evicted
+}
+
+func (m *matcherCacheMetrics) Collect(out chan<- prometheus.Metric) {
+	gm, err := m.r.Gather()
+	if err != nil {
+		level.Warn(util_log.Logger).Log("msg", "failed to gather metrics from registry", "err", err)
+		return
+	}
+
+	mfm, err := util.NewMetricFamilyMap(gm)
+
+	if err != nil {
+		level.Warn(util_log.Logger).Log("msg", "failed to create metric family map", "err", err)
+		return
+	}
+
+	out <- prometheus.MustNewConstMetric(m.requestsTotal, prometheus.CounterValue, mfm.SumCounters("thanos_matchers_cache_requests_total"))
+	out <- prometheus.MustNewConstMetric(m.hitsTotal, prometheus.CounterValue, mfm.SumCounters("thanos_matchers_cache_hits_total"))
+	out <- prometheus.MustNewConstMetric(m.numItems, prometheus.GaugeValue, mfm.SumGauges("thanos_matchers_cache_items"))
+	out <- prometheus.MustNewConstMetric(m.maxItems, prometheus.GaugeValue, mfm.SumGauges("thanos_matchers_cache_max_items"))
+	out <- prometheus.MustNewConstMetric(m.evicted, prometheus.CounterValue, mfm.SumCounters("thanos_matchers_cache_max_items"))
```

Comment:
Seems the wrong metric here. Should be evict

Reply:
good catch! Creating a test around it.